jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/DiskRule/CheckDiskEncryption.java

```diff
@@ -120,17 +120,15 @@ private boolean validateDiskEncryption(String esUrl, Map<String, Object> mustFil
                         .get("diskInner").getAsJsonObject();
                 if (diskInner != null) {
                     if (!isAttachedToVirtualMachine) {
-                        JsonObject encryptionSettings = diskInner.getAsJsonObject()
-                                .get("properties.encryptionSettings").getAsJsonObject();
-
-                        if (encryptionSettings == null) {
+                        if (diskInner.getAsJsonObject().get("properties.encryptionSettingsCollection").isJsonNull()) {
                             validationResult = false;
                         } else {
-                            boolean isEnabled = jsonDataItem.getAsJsonObject()
+                            JsonObject encryptionSettings = diskInner.getAsJsonObject()
+                                    .get("properties.encryptionSettingsCollection").getAsJsonObject();
+                            boolean isEnabled = encryptionSettings.getAsJsonObject()
```

Comment:
It is already JsonObject - you casted it already in the line above, right?

Reply:
Changed -> jsonDataItem to encryptionSettings
Example of azure disk JSON:

{"discoverydate":"2023-10-26 11:00:00+0000","_cloudType":"Azure","subscription":"f4d319d8-7eac-4e15-a561-400f7744aa81","region":"centralus","subscriptionName":"dev-paladincloud","resourceGroupName":"DEV-PALADINCLOUD","id":"subscriptions/f4d319d8-7eac-4e15-a561-400f7744aa81/resourceGroups/DEV-PALADINCLOUD/providers/Microsoft.Compute/disks/testing_disk1_82e2c9d0e28746c18f86e7dc52694290","name":"testing_disk1_82e2c9d0e28746c18f86e7dc52694290","isAttachedToVirtualMachine":true,"key":"c8d44b2d-803c-4900-9de5-0daa92f38392","diskInner":{"location":"centralus","tags":{"Application":"Jupiter","Environment":"qa"},"sku":{"name":"Premium_LRS"},"zones":null,"extendedLocation":null,"properties.osType":"Linux","properties.hyperVGeneration":"V2","properties.purchasePlan":null,"properties.creationData":{"createOption":"FromImage","storageAccountId":null,"imageReference":{"id":"/Subscriptions/6c34daa5-4db3-42a7-880d-29fc999ec966/Providers/Microsoft.Compute/Locations/centralus/Publishers/canonical/ArtifactTypes/VMImage/Offers/0001-com-ubuntu-server-focal/Skus/20_04-lts-gen2/Versions/20.04.202206060","lun":null},"galleryImageReference":null,"sourceUri":null,"sourceResourceId":null,"uploadSizeBytes":null,"logicalSectorSize":null},"properties.diskSizeGB":30,"properties.encryptionSettingsCollection":{"enabled":true,"encryptionSettings":[{"diskEncryptionKey":{"sourceVault":{"id":"/subscriptions/f4d319d8-7eac-4e15-a561-400f7744aa81/resourceGroups/dev-paladincloud/providers/Microsoft.KeyVault/vaults/paladinvault"},"secretUrl":"https://paladinvault.vault.azure.net/secrets/d066bf61-629c-44d0-99ef-e0a43ce04802/1856a8a11f504b2786a3e9d61793a99f"},"keyEncryptionKey":{"sourceVault":{"id":"/subscriptions/f4d319d8-7eac-4e15-a561-400f7744aa81/resourceGroups/dev-paladincloud/providers/Microsoft.KeyVault/vaults/paladinvault"},"keyUrl":"https://paladinvault.vault.azure.net/keys/testing/102d76bc04e94ca88ae25e04620318c2"}}],"encryptionSettingsVersion":"1.1"},"properties.diskIOPSReadWrite":120,"properties.diskMBpsReadWrite":25,"properties.diskIOPSReadOnly":null,"properties.diskMBpsReadOnly":null,"properties.diskState":"Reserved","properties.encryption":{"diskEncryptionSetId":null,"type":"EncryptionAtRestWithPlatformKey"},"properties.maxShares":null,"properties.networkAccessPolicy":"AllowAll","properties.diskAccessId":null,"properties.tier":"P4","properties.burstingEnabled":null},"sizeInGB":30,"type":"Microsoft.Compute/disks","virtualMachineId":"/subscriptions/f4d319d8-7eac-4e15-a561-400f7744aa81/resourceGroups/dev-paladincloud/providers/Microsoft.Compute/virtualMachines/testing","tags":{"Application":"Jupiter","Environment":"qa"}}